core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1151,6 +1144,12 @@ class LogManager(logDirs: Seq[File],
       }
     }
   }
+
+  private def removeLogAndMetrics(logs: Pool[TopicPartition, Log], tp: TopicPartition): Log = {
+    val removedLog = logs.remove(tp)
+    if (removedLog != null) removedLog.removeLogMetrics()
+    removedLog
```

Comment:
nit: Lets return Option(removedLog) to ease null checking by clients.

Seems like the same object gets returned by `asyncDelete` but is only used in one place in test code, so many want to change the return value of that too. The less "null" we have the better.